src/test/regress/expected/multi_router_planner.out

```diff
@@ -0,0 +1,634 @@
+ERROR:  cannot perform distributed planning on this query
+DETAIL:  Complex table expressions are currently unsupported
+-- subqueries are not supported in WHERE clause in Citus
+SELECT * FROM articles_hash WHERE author_id IN (SELECT id FROM authors_hash WHERE name LIKE '%a');
+ERROR:  cannot plan queries that include both regular and partitioned relations
+-- subqueries are supported in FROM clause but they are not router plannable
+SELECT articles_hash.id,test.word_count
+FROM articles_hash, (SELECT id, word_count FROM articles_hash) AS test WHERE test.id = articles_hash.id
+ORDER BY articles_hash.id;
+DEBUG:  join prunable for task partitionId 0 and 1
+DEBUG:  join prunable for task partitionId 0 and 2
+DEBUG:  join prunable for task partitionId 0 and 3
+DEBUG:  join prunable for task partitionId 1 and 0
+DEBUG:  join prunable for task partitionId 1 and 2
+DEBUG:  join prunable for task partitionId 1 and 3
+DEBUG:  join prunable for task partitionId 2 and 0
+DEBUG:  join prunable for task partitionId 2 and 1
+DEBUG:  join prunable for task partitionId 2 and 3
+DEBUG:  join prunable for task partitionId 3 and 0
+DEBUG:  join prunable for task partitionId 3 and 1
+DEBUG:  join prunable for task partitionId 3 and 2
+DEBUG:  pruning merge fetch taskId 1
+DETAIL:  Creating dependency on merge taskId 5
+DEBUG:  pruning merge fetch taskId 2
+DETAIL:  Creating dependency on merge taskId 5
+DEBUG:  pruning merge fetch taskId 4
+DETAIL:  Creating dependency on merge taskId 8
+DEBUG:  pruning merge fetch taskId 5
+DETAIL:  Creating dependency on merge taskId 8
+DEBUG:  pruning merge fetch taskId 7
+DETAIL:  Creating dependency on merge taskId 11
+DEBUG:  pruning merge fetch taskId 8
+DETAIL:  Creating dependency on merge taskId 11
+DEBUG:  pruning merge fetch taskId 10
+DETAIL:  Creating dependency on merge taskId 14
+DEBUG:  pruning merge fetch taskId 11
+DETAIL:  Creating dependency on merge taskId 14
+ERROR:  cannot use real time executor with repartition jobs
+HINT:  Set citus.task_executor_type to "task-tracker".
+-- subqueries are not supported in SELECT clause
+SELECT a.title AS name, (SELECT a2.id FROM authors_hash a2 WHERE a.id = a2.id  LIMIT 1)
+						 AS special_price FROM articles_hash a;
+ERROR:  cannot perform distributed planning on this query
+DETAIL:  Subqueries other than in from-clause are currently unsupported
+set citus.task_executor_type to 'router';
+-- simple lookup query
+SELECT *
+	FROM articles_hash
+	WHERE author_id = 1;
+DEBUG:  Creating router plan
+DEBUG:  predicate pruning for shardId 103201
+ id | author_id |    title     | word_count 
+----+-----------+--------------+------------
+  1 |         1 | arsenous     |       9572
+ 11 |         1 | alamo        |       1347
+ 21 |         1 | arcading     |       5890
+ 31 |         1 | athwartships |       7271
+ 41 |         1 | aznavour     |      11814
+(5 rows)
+
+-- below query hits a single shard, but it is not router plannable
```

Comment:
Oh I see, we do `+set citus.task_executor_type to 'router';`. But, I think the bigger question is, do we want to keep the config?
There a very few cases that router executor can execute queries that router executor not planned such as:
* `WHERE part_col = val1 OR part_cal = val` if this hits to a single shard
* Selfs joins 'WHERE a.part_col=val and a.part_col=b.part_cal', which is not a very useful query.

What I think is that it'd be confusing to have that GUC and no one would be using it. What do you think?